### PR TITLE
rust type generation: fix public + hashmap import

### PR DESCRIPTION
**Pull request by Viterkim**

### CODE REVIEW

**Comment by Viterkim** [comment]
Ready now

File: src/validators/object.ts [hunk 1]
```diff
@@ -173,36 +176,60 @@ export abstract class ObjectValidator<T extends ObjectSchema = never, O = never>
           }
         }
 
+        // Serde
+        let serdeStr: string
+        let deriveMacro: string[] | undefined = undefined
+
+        if (this.deriveMacro !== undefined) {
```
Comment (tlbdk) [inline src/validators/object.ts:183]: Non of the other options are rust specefic, why is this one needed, if I look at the examples it looks like deriveMacro is always the same expect for one case where Debug is not set, does it really matter if that is not set?

File: src/validators/object.ts [hunk 2]
```diff
@@ -173,36 +176,60 @@ export abstract class ObjectValidator<T extends ObjectSchema = never, O = never>
...
+          const deriveMacroStr = this.deriveMacro.join(', ')
+          serdeStr = `#[derive(${deriveMacroStr})]\n`
+          deriveMacro = this.deriveMacro
+        } else if (options.deriveMacro !== undefined) {
```
Comment (tlbdk) [inline src/validators/object.ts:187]: Are you not setting ```this.deriveMacro = options?.deriveMacro``` in the constructor and you are setting this.deriveMacro if options.deriveMacro !== undefined, don't think this code block makes sense.

File: src/validators/object.ts (hunk 1 above)
Comment (Viterkim) [reply src/validators/object.ts:183]: Yes you want Eq, PartialEq or other stuff sometimes. (which you cannot do for all types)

Comment (Viterkim) [reply src/validators/object.ts:183]: To make matters worse, to be backwards compatible, we might need a custom field/macro above each field in an object/enum. #[serde(SOmething)] etc

File: src/validators/object.ts (hunk 2 above)
Comment (Viterkim) [reply src/validators/object.ts:187]: There's the deriveMacro from "options" which would be the the call toString() where the parrent can pass in stuff.
And then there's the constructor/class field, which takes priority. (So manually writing it).

**Review by tlbdk — APPROVED** [review]

**Pull request merged**